pkg/kubelet/container/sync_result.go

```diff
@@ -19,14 +19,62 @@ package container
 import (
 	"errors"
 	"fmt"
+	"time"
 
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 )
 
 // TODO(random-liu): We need to better organize runtime errors for introspection.
 
-// ErrCrashLoopBackOff returned when a container Terminated and Kubelet is backing off the restart.
-var ErrCrashLoopBackOff = errors.New("CrashLoopBackOff")
+// BackoffError should be used whenever an error needs to specify a particular backoff duration
+// to the Kubelet.
+type BackoffError struct {
+	err         error
+	backoffTime time.Duration
+}
+
+func NewBackoffError(err error, backoffTime time.Duration) *BackoffError {
+	return &BackoffError{
+		err:         err,
+		backoffTime: backoffTime,
+	}
+}
+
+func (e *BackoffError) Error() string {
+	return e.err.Error()
+}
+
+func (e *BackoffError) BackoffTime() time.Duration {
+	return e.backoffTime
+}
+
+// MinBackoff recursively searches through err for BackoffErrors and returns
+// the minimum of all found backoff durations.
+func MinBackoff(err error) (time.Duration, bool) {
+	var ae utilerrors.Aggregate
+	var be *BackoffError
+	switch {
+	case errors.As(err, &be):
+		return be.BackoffTime(), true
+	case errors.As(err, &ae):
+		var min time.Duration
+		found := false
+		for _, e := range ae.Errors() {
+			if backoff, ok := MinBackoff(e); ok {
+				if !found || backoff < min {
+					min = backoff
+					found = true
+				}
+			}
+		}
+		return min, found
+	default:
+		if e := errors.Unwrap(err); e != nil {
+			return MinBackoff(e)
+		}
```

Comment:
I don't think you need to call `Unwrap` here, since `errors.As` already does that.